--- openvpn_defer_auth.c.orig	2018-06-29 19:42:59.000000000 +0000
+++ openvpn_defer_auth.c	2018-06-29 20:20:11.000000000 +0000
@@ -32,6 +32,7 @@
  */
 
 struct plugin_context {
+    char *script_path;
 };
 
 OPENVPN_EXPORT openvpn_plugin_handle_t
@@ -45,6 +46,14 @@
      */
     context = (struct plugin_context *) calloc(1, sizeof(struct plugin_context));
 
+    if (argv[1]) {
+        context->script_path = strdup(argv[1]);
+        if (context->script_path == NULL) {
+            perror("Unable to allocate memory\n");
+            exit(EXIT_FAILURE);
+        }
+    }
+
     /*
      * Which callbacks to intercept.
      */
@@ -83,5 +92,6 @@
 openvpn_plugin_close_v1(openvpn_plugin_handle_t handle)
 {
     struct plugin_context *context = (struct plugin_context *) handle;
+    free(context->script_path);
     free(context);
 }
