--- openvpn_defer_auth.c.orig	2018-06-29 19:14:27.000000000 +0000
+++ openvpn_defer_auth.c	2018-06-29 19:13:04.000000000 +0000
@@ -131,11 +131,12 @@
 }
 
 OPENVPN_EXPORT openvpn_plugin_handle_t
-openvpn_plugin_open_v1(unsigned int *type_mask, const char *argv[], const char *envp[])
+openvpn_plugin_open_v2(unsigned int *type_mask, const char *argv[], const char *envp[],
+						struct openvpn_plugin_string_list **return_list)
 {
     struct plugin_context *context;
 
-    printf("FUNC: openvpn_plugin_open_v1\n");
+    printf("FUNC: openvpn_plugin_open_v2\n");
 
     /*
      * Allocate our context
