--- openvpn_defer_auth.c.orig	2019-04-18 23:44:23.469292428 +0000
+++ openvpn_defer_auth.c	2019-04-18 23:46:22.727335388 +0000
@@ -31,6 +31,11 @@
 #include "openvpn-plugin.h"
 
 /*
+ * Constants
+ */
+#define OPENVPN_PLUGIN_VERSION_MIN 3
+
+/*
  * Our context, where we keep our state.
  */
 
@@ -46,6 +51,15 @@
     while(waitpid((pid_t)(-1), 0, WNOHANG) > 0) {}
 }
 
+/*
+ * Require OpenVPN Plugin API v3
+ */
+OPENVPN_EXPORT int
+openvpn_plugin_min_version_required_v1()
+{
+    return OPENVPN_PLUGIN_VERSION_MIN;
+}
+
 OPENVPN_EXPORT openvpn_plugin_handle_t
 openvpn_plugin_open_v2(unsigned int *type_mask, const char *argv[], const char *envp[],
 						struct openvpn_plugin_string_list **return_list)
